ipapython/ntpmethods.py

```diff
@@ -0,0 +1,243 @@
+#
+# Copyright (C) 2018  FreeIPA Contributors see COPYING for license
+#
+from __future__ import absolute_import
+
+import shutil
+import re
+import sys
+
+# pylint: disable=import-error,ipa-forbidden-import
+from ipaclient.install import ipadiscovery  # pylint: disable=E0401,E0611
+# pylint: enable=import-error,ipa-forbidden-import
+from ipaplatform import services
+from ipaplatform.paths import paths
+from ipaplatform.constants import constants
+from ipapython import ipautil
+
+
+def service_command():
+    timedata_srv = {
+        'openntpd': {
+            'api': services.knownservices.openntpd,
+            'service': 'openntpd',
+        },
+        'ntpd': {
+            'api': services.knownservices.ntpd,
+            'service': 'ntpd',
+        },
+        'chrony': {
+            'api': services.knownservices.chronyd,
+            'service': 'chronyd'
+        }
+    }
+
+    return timedata_srv[detect_time_server()]
+
+
+def detect_time_server():
+    ts_modules = ['chrony', 'ntpd', 'openntpd']
+
+    for ts in ts_modules:
+        sys_ts = ipautil.run(
+            [paths.PACKAGE_MANAGER, constants.CHECK_PACKAGE_OPT, ts],
+            capture_output=True, raiseonerr=False)
+        if sys_ts.output:
+            return ts
+
+    print("NTP daemon not found in your system. "
+          "Please, install NTP daemon and try again.")
+    sys.exit(1)
+
+
+def search_ntp_servers(statestore, cli_domain):
+    force_service(statestore)
+    ds = ipadiscovery.IPADiscovery()
+    ntp_servers = ds.ipadns_search_srv(
+        cli_domain,
+        '_ntp._udp',
+        None, False
+    )
+
+    return ntp_servers
+
+
+def backup_config(ntp_confile, fstore=None):
+    if fstore:
+        fstore.backup_file(ntp_confile)
+    else:
+        shutil.copy(ntp_confile, "%s.ipasave" % ntp_confile)
+
+
+def write_config(ntp_confile, content):
+    fd = open(ntp_confile, "w")
+    fd.write(content)
+    fd.close()
+
+
+def restore_state(statestore, fstore, ntp_confile, logger):
+    try:
+        fstore.restore_file(ntp_confile)
+    except ValueError:
+        logger.debug("Configuration file %s was not restored.", ntp_confile)
+
+    service_command()['api'].stop()
+    service_command()['api'].disable()
+
+    if statestore:
+        enabled = statestore.restore_state(service_command()['service'],
+                                           'enabled')
+        running = statestore.restore_state(service_command()['service'],
+                                           'running')
+
+        if enabled:
+            service_command()['api'].enable()
+
+        if running:
+            service_command()['api'].start()
+
+
+def check_timedate_services():
+    for service in services.timedate_services:
+        if service == service_command()['service']:
```

Comment:
I believe that this line is causing deprecation warnings:
```
/usr/lib/python3.6/site-packages/ipaplatform/base/services.py:211: RuntimeWarning: RedHatService('ntpd', api=None) is deprecated.
  super(SystemdService, self).__init__(service_name, api=api)
/usr/lib/python3.6/site-packages/ipaplatform/base/services.py:211: RuntimeWarning: RedHatService('openntpd', api=None) is deprecated.
  super(SystemdService, self).__init__(service_name, api=api)
```

Reply:
This happens after I run ipa-server-install on fedora. What system do you use for testing this?